pkg/node/node.go

```diff
@@ -1665,89 +1638,527 @@ func (s *Service) buildInitiatorsArray(initiators []string, arrayID string) []go
 }
 
 // create or update host on PowerStore array
-func (s *Service) updateHost(ctx context.Context, initiators []string, client gopowerstore.Client, host gopowerstore.Host, arrayID string) (err error) {
+func (s *Service) updateHost(ctx context.Context, initiators []string, client gopowerstore.Client, host gopowerstore.Host, arrayID string, connectivity *gopowerstore.HostConnectivityEnum) error {
 	initiatorsToAdd, initiatorsToDelete := checkIQNS(initiators, host)
-	return s.modifyHostInitiators(ctx, host.ID, client, initiatorsToAdd, initiatorsToDelete, nil, arrayID)
+
+	return s.modifyHostInitiators(ctx, host.ID, client, initiatorsToAdd, initiatorsToDelete, nil, arrayID, connectivity)
 }
 
+// getNodeLabels retrieves the kubernetes node object and returns its labels
+func (s *Service) getNodeLabels(nodeName string) (map[string]string, error) {
+	return k8sutils.GetNodeLabels(context.Background(), s.opts.KubeConfigPath, nodeName)
+}
+
+var (
+	getNodeLabelsfn = func(s *Service, kubeNodeName string) (map[string]string, error) {
+		return s.getNodeLabels(kubeNodeName)
+	}
+
+	getArrayfn = func(s *Service) map[string]*array.PowerStoreArray {
+		return s.Arrays()
+	}
+
+	getIsHostAlreadyRegistered = func(s *Service, ctx context.Context, client gopowerstore.Client, initiators []string) bool {
+		return s.isHostAlreadyRegistered(ctx, client, initiators)
+	}
+
+	getAllRemoteSystemsFunc = func(arr *array.PowerStoreArray, ctx context.Context) ([]gopowerstore.RemoteSystem, error) {
+		return arr.GetClient().GetAllRemoteSystems(ctx)
```

Comment:
It should be possible to mock gopowerstore calls using a mock Client.

Reply:
This code is covered with the mocks.